Feat/Connect password reset function with frontend

bank1e opened the pull request:
**Purpose:**
- Create UI to navigate user to forgot password page.
- Create UI to initiate password reset function through the URL that user received.

**Change(s):**
- Installed query-string to take token from password reset URL.
- Hard coded local server path in authController. (Reason: It needs the client's local path, not backend local path.)
- Added ForgotPasswordPage and ResetPasswordPage in App.jsx.
- Created new components for Forgot/Reset Password Pages.
- Added link to ForgotPasswordPage in LoginPage.jsx.

**How to review:**
![Video_2024-05-31_164658](https://github.com/chingu-voyages/v49-tier3-team-29/assets/166264656/ca6970f6-244f-4c23-816c-0c79ffe6953d)

1. Navigate to click "Forgot Password" link.
2. Input an existing user's email address that you can receive password reset email.
3. If email address exists in database, it sends email to reset password. If not, error "User not found" pops up.
4. Click the URL in the email.
5. Input a new password.
6. Try to log in with the new password you just set, and confirm the success response.

![Video_2024-05-31_171923](https://github.com/chingu-voyages/v49-tier3-team-29/assets/166264656/54f0cb44-fdf0-4d1c-9718-b782a035b7b5)
👆️ Error:  If token is invalid or expired, an error shows up. - _Invalid or expired password reset token_ 

![Video_2024-05-31_172127](https://github.com/chingu-voyages/v49-tier3-team-29/assets/166264656/97c7d556-a1c8-43e3-841a-429e176f5164)
👆️ Error: Email address does not match. - _User not found_

alexh205 reviewed (commented):
- Please add a button to both the Forgot/Reset Password pages to trigger the form submission (as it took me some time initially to initiate the submission). Having visual buttons will provide a more user-friendly functionality to the components. Retain the current trigger and only add the buttons as a secondary trigger. 

- The Reset Password page should have a secondary input field "Verify New Password". This will allow users the ability to confirm the accuracy of their password entry before saving it to the Database. 

Please do let me know if you have any questions.

bank1e commented:
**Screenshots:**
![image](https://github.com/chingu-voyages/v49-tier3-team-29/assets/166264656/b718b05f-f541-4d89-a931-de73bc92db73)
![image](https://github.com/chingu-voyages/v49-tier3-team-29/assets/166264656/fa8d6793-3f4d-4e41-95d8-9686cc929190)
![image](https://github.com/chingu-voyages/v49-tier3-team-29/assets/166264656/2cdc7d29-28da-4564-a88f-6c9c2a4792fa)
![image](https://github.com/chingu-voyages/v49-tier3-team-29/assets/166264656/84254570-b4a4-4ff8-81c1-d49d3b9a0612)

**Changes:**
1.  Added Submit button in forgot/reset password page.
2.  Added minimum length validation in text field in reset password page.
3.  Added password repeat field and validation in reset password page.

bank1e commented:
Thank you for the comments! Please review the changes again.

bank1e commented:
![image](https://github.com/chingu-voyages/v49-tier3-team-29/assets/166264656/66608615-762f-4e66-8311-4aef83a8ccab)
Added "Repeat password" message in the text field in password reset page.

alexh205 reviewed (commented):
The updates look great!

Here is one minor update: 
- Currently, within the 'ResetPasswordPage.jsx' module, the 'onSubmit' function will trigger 'fetchResetPasswordAPI' even if the 'newPassword' and 'newPasswordVerify' parameters do not match. If you place the 'fetchResetPasswordApi' function with an 'ELSE' block, then that should fix the issue. 


![image](https://github.com/chingu-voyages/v49-tier3-team-29/assets/95322089/54776769-e108-44d3-95a1-a11da9b5742d)

- The two passwords are of different lengths, so the first alert is correct

![image](https://github.com/chingu-voyages/v49-tier3-team-29/assets/95322089/6d3ea391-8da5-49d2-ac9e-549e5b7902e3)
- Once the first alert 'Passwords do not match' is confirmed, the secondary alert 'Password reset successful' is incorrectly triggered. 

![image](https://github.com/chingu-voyages/v49-tier3-team-29/assets/95322089/b8980851-d2fc-4796-b2a1-56f4335d3ce8)
 - This is the solution to address the incorrect function trigger.


All other functionality works well.

bank1e commented on client/src/components/ResetPasswordPage/ResetPasswordPage.jsx:
```diff
@@ -0,0 +1,122 @@
+    event.preventDefault();
+    if (newPassword !== newPasswordVerify) {
+      alert("Passwords do not match");
+    } else {
```
Thank you for the detailed advise! It is very helpful for correction. 

- As advised, **else** is inserted before fetchResetPasswordApi. 
- Conflict in **App.jsx** has been resolved.

alexh205 reviewed (approved):
Approved
